packages/cells/src/widget.ts

```diff
@@ -1445,8 +1460,8 @@ export class MarkdownCell extends AttachmentsCell {
   /**
    * Modify the cell source to include a reference to the attachment.
    */
-  protected updateCellSourceWithAttachment(attachmentName: string) {
-    const textToBeAppended = `![${attachmentName}](attachment:${attachmentName})`;
+  protected updateCellSourceWithAttachment(attachmentName: string, URI?: string) {
+    const textToBeAppended = `![${attachmentName}](attachment:${URI ?? encodeURI(attachmentName)})`;
```

Comment:
On second thought, encoding the `attachmentName` changes the behavior in a backwards-incompatible way, so I'll revert my suggestion before merging.
```suggestion
    const textToBeAppended = `![${attachmentName}](attachment:${URI ?? attachmentName})`;
```